test/AngularSpec.js

```diff
@@ -602,6 +602,21 @@ describe('angular', function() {
       expect(copy(new Number(NaN)).valueOf()).toBeNaN();
       /* eslint-enable */
     });
+
+    it('should copy source until reaching a given max depth', function() {
+      var destinationA = {};
+      var destinationB = {};
+
+      destinationA = copy({a1:1, b1:{b2:{b3:1}}, c1:[1,{c2:1}], d1:{d2:1}}, destinationB, 1);
+      expect(destinationA).toEqual({a1:1, b1:'...', c1:'...', d1:'...'});
+      expect(destinationB).toEqual({a1:1, b1:'...', c1:'...', d1:'...'});
+
+      destinationA = {};
+      destinationB = {};
+      destinationA = copy({a1:1, b1:{b2:{b3:1}}, c1:[1,{c2:1}], d1:{d2:1}}, destinationB, 2);
+      expect(destinationA).toEqual({a1:1, b1:{b2:'...'}, c1:[1,'...'], d1:{d2:1}});
+      expect(destinationB).toEqual({a1:1, b1:{b2:'...'}, c1:[1,'...'], d1:{d2:1}});
+    });
```

Comment:
Add a few more tests to cover more cases. E.g.:

- `maxDepth: 4`
- `maxDepth: 0`
- `maxDepth: -1`
- `maxDepth: NaN`
- `maxDepth: true`
- `maxDepth: null`
- no `maxDepth`